pandas/core/groupby/groupby.py

```diff
@@ -2666,12 +2695,18 @@ def _get_cythonized_result(
 
         labels, _, ngroups = grouper.group_info
         output: Dict[base.OutputKey, np.ndarray] = {}
-        base_func = getattr(libgroupby, how)
+        if cython_dtype is not None:
+            base_func = getattr(libgroupby, how)
 
         error_msg = ""
         for idx, obj in enumerate(self._iterate_slices()):
             name = obj.name
             values = obj._values
+            if cython_dtype is None:
+                cython_dtype = values.dtype
+                # We also need to get the specific function for that dtype
+                how += f"_{cython_dtype}"
```

Comment:
using cython fused types should make cython figure this out for us (also avoids need for tempita)

Reply:
I have to use tempita because of the hashtables have different functions based on dtype.